core_helpers/selenium_action_objects.py

```diff
@@ -129,11 +129,14 @@ def set_text(self,locator,value,clear_flag=True):
 
         return result_flag
 
-    def get_text(self,locator):
+    def get_text(self,locator, dom_element_flag=False):
         "Return the text for a given path or the 'None' object if the element is not found"
         text = ''
         try:
-            text = self.get_element(locator).text
+            if dom_element_flag is False:
```

Comment:
Why we need dom_element_flag ? under else part, Is locator.text returns text. Asking because mostly our locator contains (selector_type, xpath/css).

Reply:
We would need it incase we are trying to extract the text directly from DOM elements. For example, we fetch all the product names in a page with 'get_elements', and then if we want to extract the product name of each individual elements from that list, we can directly use the get_text method by setting the DOM_flag to true.
It's also useful when any method returns a DOM element that we need to manipulate